cmd/kops/create_cluster.go

```diff
@@ -1156,43 +1165,44 @@ func RunCreateCluster(f *util.Factory, out io.Writer, c *CreateClusterOptions) e
 		return fmt.Errorf("error writing completed cluster spec: %v", err)
 	}
 
-	if len(c.SSHPublicKeys) == 0 {
-		autoloadSSHPublicKeys := true
-		switch c.Cloud {
-		case "gce":
-			// We don't normally use SSH keys on GCE
-			autoloadSSHPublicKeys = false
-		}
+	if !c.NoSSHKey {
+		if len(c.SSHPublicKeys) == 0 {
+			autoloadSSHPublicKeys := true
+			switch c.Cloud {
+			case "gce":
+				// We don't normally use SSH keys on GCE
+				autoloadSSHPublicKeys = false
+			}
 
-		if autoloadSSHPublicKeys {
-			// Load from default location, if found
-			sshPublicKeyPath := "~/.ssh/id_rsa.pub"
-			c.SSHPublicKeys, err = loadSSHPublicKeys(sshPublicKeyPath)
-			if err != nil {
-				// Don't wrap file-not-found
-				if os.IsNotExist(err) {
-					glog.V(2).Infof("ssh key not found at %s", sshPublicKeyPath)
-				} else {
-					return fmt.Errorf("error reading SSH key file %q: %v", sshPublicKeyPath, err)
+			if autoloadSSHPublicKeys {
+				// Load from default location, if found
+				sshPublicKeyPath := "~/.ssh/id_rsa.pub"
+				c.SSHPublicKeys, err = loadSSHPublicKeys(sshPublicKeyPath)
+				if err != nil {
+					// Don't wrap file-not-found
+					if os.IsNotExist(err) {
+						glog.V(2).Infof("ssh key not found at %s", sshPublicKeyPath)
+					} else {
+						return fmt.Errorf("error reading SSH key file %q: %v", sshPublicKeyPath, err)
+					}
 				}
 			}
 		}
-	}
-
-	if len(c.SSHPublicKeys) != 0 {
-		sshCredentialStore, err := clientset.SSHCredentialStore(cluster)
-		if err != nil {
-			return err
-		}
 
-		for k, data := range c.SSHPublicKeys {
-			err = sshCredentialStore.AddSSHPublicKey(k, data)
+		if len(c.SSHPublicKeys) != 0 {
+			sshCredentialStore, err := clientset.SSHCredentialStore(cluster)
 			if err != nil {
-				return fmt.Errorf("error adding SSH public key: %v", err)
+				return err
+			}
+
+			for k, data := range c.SSHPublicKeys {
+				err = sshCredentialStore.AddSSHPublicKey(k, data)
+				if err != nil {
+					return fmt.Errorf("error adding SSH public key: %v", err)
+				}
```

Comment:
I think we move the sanity check here:

If we're on a platform that normally uses SSH keys (i.e. not GCE), and we have no ssh public keys, and `no-ssh-key == false`, then we raise an error suggesting that the user specifies a public key, or passes the `no-ssh-key` flag